oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RulesReader.java

```diff
@@ -49,93 +50,136 @@ public Rules readRules() {
-                        alarmRule.setIncludeNamesRegex((String) settings.getOrDefault("include-names-regex", ""));
-                        alarmRule.setExcludeNamesRegex((String) settings.getOrDefault("exclude-names-regex", ""));
-                        alarmRule.setIncludeLabels(
-                            (ArrayList) settings.getOrDefault("include-labels", new ArrayList(0)));
-                        alarmRule.setExcludeLabels(
-                            (ArrayList) settings.getOrDefault("exclude-labels", new ArrayList(0)));
-                        alarmRule.setIncludeLabelsRegex((String) settings.getOrDefault("include-labels-regex", ""));
-                        alarmRule.setExcludeLabelsRegex((String) settings.getOrDefault("exclude-labels-regex", ""));
-                        alarmRule.setThreshold(settings.get("threshold").toString());
-                        alarmRule.setOp((String) settings.get("op"));
-                        alarmRule.setPeriod((Integer) settings.getOrDefault("period", 1));
-                        alarmRule.setCount((Integer) settings.getOrDefault("count", 1));
-                        // How many times of checks, the alarm keeps silence after alarm triggered, default as same as period.
-                        alarmRule.setSilencePeriod((Integer) settings.getOrDefault("silence-period", alarmRule.getPeriod()));
-                        alarmRule.setMessage(
-                            (String) settings.getOrDefault("message", "Alarm caused by Rule " + alarmRule
-                                .getAlarmRuleName()));
+            readRulesConfig(rules);
+            readCompositeRuleConfig(rules);
+            readWebHookConfig(rules);
+            readGrpcConfig(rules);
+            readSlackConfig(rules);
+            readWechatConfig(rules);
+        }
+        return rules;
+    }
 
-                        rules.getRules().add(alarmRule);
-                    }
-                });
-            }
-            List webhooks = (List) yamlData.get("webhooks");
-            if (webhooks != null) {
-                rules.setWebhooks(new ArrayList<>());
-                webhooks.forEach(url -> {
-                    rules.getWebhooks().add((String) url);
-                });
+    private void readCompositeRuleConfig(Rules rules) {
+        Map compositeRulesData = (Map) yamlData.get("composite-rules");
+        if (compositeRulesData == null) {
+            return;
+        }
+        compositeRulesData.forEach((k, v) -> {
+            String ruleName = (String) k;
+            if (ruleName.endsWith("_rule")) {
+                Map settings = (Map) v;
+                CompositeAlarmRule compositeAlarmRule = new CompositeAlarmRule();
+                compositeAlarmRule.setAlarmRuleName(ruleName);
+                String expression = (String) settings.get("expression");
+                if (expression == null) {
+                    throw new IllegalArgumentException("expression can't be null");
```

Comment:
Could you recheck what is the status if the dynamic config center pushing an illegal alarm setting?

Reply:
This will throws a exception, and current new rules will not effect, the system will use old rules config, 
do you mean I should not throws Exception ?

Reply:
I didn't mean anything should be changed, I was just rechecking the status. It is good enough

Reply:
got it